### PR TITLE
kvserver: add overfull no rebalance options metric

### PR DESCRIPTION
**Pull request by kvoli**
Previously, there was no metric indication that a store was unable to
reduces its load below a balance threshold due to exhausing potential
rebalance actions. This metric is desirable to determine when the action
space needs to be increased (by splitting etc) to balance load.

This patch adds this metric:

`rebalancing.rebalancing.state.imbalanced_overfull_options_exhausted`

Which maintains a counter, incremented each time the a store's
rebalancer is unable to reduce the load on the store below the overfull
threshold due to running out of available rebalance actions.

Part of: https://github.com/cockroachdb/cockroach/issues/90582

Release note: None

### CODE REVIEW

**Review by andrewbaptist — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94931#-:-NLSByCeCnNgpac0dk0x:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kvoli)

___
*[`pkg/kv/kvserver/store_rebalancer.go` line 49 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94931#-NLSBjZG2fIkBjsPvH-u:-NLSBjZG2fIkBjsPvH-v:bpll01e) ([raw file](https://github.com/cockroachdb/cockroach/blob/54b6a1edc0cf412e8bce439fc2a1ecf1a7b4777c/pkg/kv/kvserver/store_rebalancer.go#L49)):*
> ```Go
> 	metaStoreRebalancerImbalancedOverfullOptionsExhausted = metric.Metadata{
> 		Name: "rebalancing.state.imbalanced_overfull_options_exhausted",
> 		Help: "Number of occurences where this store was imbalanced by being overfull " +
> ```

nit: occurences -> occurrences

Also, the grammar of the sentence is a little hard to follow. Consider rewording this to something like: 

"Number of occurrences where this store was overfull but failed to shed load after exhausting available rebalance options"

**Review by kvoli — COMMENTED** [review]
TYFTR

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94931#-:-NLSzA9b79ql-L4JNho4:b-80sls6)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andrewbaptist)

___
*[`pkg/kv/kvserver/store_rebalancer.go` line 49 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94931#-NLSBjZG2fIkBjsPvH-u:-NLSz7u1DAnPB_cGU3as:bh6d0zw) ([raw file](https://github.com/cockroachdb/cockroach/blob/54b6a1edc0cf412e8bce439fc2a1ecf1a7b4777c/pkg/kv/kvserver/store_rebalancer.go#L49)):*
<details><summary><i>Previously, andrewbaptist (Andrew Baptist) wrote…</i></summary><blockquote>

nit: occurences -> occurrences

Also, the grammar of the sentence is a little hard to follow. Consider rewording this to something like: 

"Number of occurrences where this store was overfull but failed to shed load after exhausting available rebalance options"
</blockquote></details>

Agreed - I updated to your description.

**Comment by kvoli** [comment]
bors r=andrewbaptist single on

**Comment by rail** [comment]
bors r-
We have temporarily disabled bors for merges. Please merge the PR manually when all required checks pass.
